### ISSUE TITLE
improve security check

### ISSUE THREAD

**Issue by tobibode** [dcooney/instant-images#55]
Hi @dcooney 

https://github.com/dcooney/instant-images/blob/17032b4eb7a3c16a304a83211b9fe19704eca5bf/api/download.php#L243
I came across this line because I'm running your plugin on a server where the SERVER_ADDR isn't set at all, that's why downloading resulted in an error.

Please also note that even if provided, the $_SERVER  superglobal can be altered e.g. by other plugins, so this check doesn't bring any real security benefit and needs to be improved or left out to avoid more users having trouble.

**Comment by tobibode** [comment]
What are you trying to avoid exactly? I'm guessing downloading images from a source other than the providers.
Can you create a list of allowed urls for every provider?

**Comment by dcooney** [comment]
> What are you trying to avoid exactly? I'm guessing downloading images from a source other than the providers.
Exactly, this was a patch put in place as an editor level vulnerability was reported. This fix was approved by the security team there.

> Can you create a list of allowed urls for every provider?
 This is how it functioned previously, but with the addition of Openverse I can't really do that because the image sources can really come from anywhere. 

I suppose I could lock down the available sources to 5 or 6 which would solve this.

**Comment by dcooney** [comment]
@tobibode FYI - The next version of the plugin removes the `$_SERVER` security check and reverts back to what I had originally which was an array of expected URLs.
![image](https://user-images.githubusercontent.com/428624/232855203-0fa103a8-0687-46a7-b444-b04d79dbda23.png)
